### ISSUE TITLE
Investigate tags and workflow versions for tools

### ISSUE THREAD

**Issue by NatalieEO** [dockstore/dockstore#2913]
In 1.6, tool versions are loaded with the label tags, but they are now loaded under the label workflowVersions. However, the tags label still exists, but it seems to always be null. Should tags still exist for tools? And does the 1.6 CLI rely on looking for tags instead of workflowVersions.

**Comment by denis-yuen** [comment]
Oh, maybe I misheard. So this isn't non-deterministic? 
That sounds like a backwards compatibility thing then (i.e. provide the old field to maintain some form of compatibility to old clients)

**Comment by NatalieEO** [comment]
You probably heard correctly. It's just when I was writing the ticket I realized what I said during standup wasn't actually the case. But thought should bring it up anyway and since I didn't quite get what gary said about what he's noticed, this might be related? Or maybe his was an ES only issue.

**Comment by denis-yuen** [comment]
I think what @garyluu meant was that when Java objects are converted and imported into elastic search, he has noticed the same thing (a rename of fields). 

So it sounds like this is similar to #2860 or https://github.com/dockstore/dockstore/pull/2887
When renaming fields, we sometimes have to leave the old one in place either with the data in the old format or just empty to avoid crashing old clients. 

Why was performance testing picking this up though? Does it rely on the old field?

**Comment by garyluu** [comment]
During standup, I noted that before my fix, elasticsearch verification facet was not working at all (completely missing).  The verification facet pulls in ALL the tools/workflows and looks for tags.verified in each entry and still found nothing which is why I believed that it's deterministically null/empty for all of them. See https://github.com/dockstore/dockstore-ui2/commit/134232f2c0e2027f20e0cb56d8b99fd8a19585f3#diff-5c2de616fd18de33171ea09ce84291d1 for me removing `tags.verified`. In the current search, it's using workflowVersions.verificationSources instead of tags.verificationSources.

**Comment by coverbeck** [comment]
> Why was performance testing picking this up though? Does it rely on the old field?

Yes, it picks a random container and then navigates to a the default version, or first version, if there is no default version, like what the UI is doing when you click on one of the containers on the home page. It is looking at the now old field to figure that out.

We can change it to check for workflowVersions instead.
